### DISCUSSION TITLE
Plasma6/Wayland: Please Select the screen to be shared.

### DISCUSSION THREAD

**Discussion by ic3man5** [rustdesk/rustdesk#7516] [General]
### Bug Description

I'm having this issue under EndeavourOS.

```
Operating System: EndeavourOS 
KDE Plasma Version: 6.0.2
KDE Frameworks Version: 6.0.0
Qt Version: 6.6.2
Kernel Version: 6.8.1-zen1-1-zen (64-bit)
Graphics Platform: Wayland
Processors: 16 × AMD Ryzen 7 5800X3D 8-Core Processor
Memory: 31.2 GiB of RAM
Graphics Processor: AMD Radeon RX 7900 XT
Manufacturer: ASUS
```

### How to Reproduce

Attempt to connect when host is EndeavourOS running KDE 6.

### Expected Behavior

To work.

### Operating system(s) on local side and remote side

EndeavourOS -> EndeavourOS

### RustDesk Version(s) on local side and remote side

1.2.3 -> 1.2.3

### Screenshots

Client: The screenshot on the client side doesn't show up long enough to get a screenshot.
Host: Dialog Screen Sharing with options for Full Workspace, New Virtual Ouput, Rectangular Region, All three monitors.

**Comment by fufesou** [comment]
@ic3man5 Hi, thanks for your feedback.

Do you mean that the Host side does not pop up the selection dialog, and you want it to pop up?

**Comment by ic3man5** [comment]
@fufesou the host side pops up every connection attempt and is never successful. This was painful because I was remote when this happened and had to have my wife select it just to have it not work.

**Reply by fufesou** [reply]
Can you please try the remember option on the Host side?

The following screenshot is ArchLinux. Your EndeavourOS may have the similar option.

<img width="292" alt="1711420639243" src="https://github.com/rustdesk/rustdesk/assets/13586388/c62c8502-5ead-44cf-9b83-b1ff90e24a79">

**Reply by ic3man5** [reply]
I will try this tomorrow, I'd consider this a pretty serious bug if this is the fix.

Edit: my wife said it was checked automatically

**Reply by fufesou** [reply]
Sorry I cannot understand you clearly.

Do you mean you have to select the monitor every connection event this option is checked?

> select it just to have it not work

"have it not work" means ?


The remember options works on my Ubuntu and Archlinux.

I'll test EndeavourOS (2024.01.25) with the default desktop.

**Reply by fufesou** [reply]
https://github.com/rustdesk/rustdesk/assets/13586388/31e535b6-7a2b-475a-aee7-04b7bf802928

**Reply by xtyuns** [reply]
after lock the screen, the remember option will not work

**Reply by ic3man5** [reply]
Sorry I wasn't able to get back until now. This doesn't seem to be directly related to rustdesk itself but KDE specifically. OBS and discord have the same exact issue, where some options are "invalid" and it bounces it back to the Screen Sharing dialog. Discord is just as broken as rustdesk and OBS seems to do a little better job. I will attempt to get a capture going later to show exactly what is going on.

I feel like this is more of an upstream issue with wherever the screen share dialog is located (KDE?).

**Reply by xlionjuan** [reply]
Wayland is born for security, but a little "to secure" if you want unattended access, and the permission control isn't complete.

**Reply by giarve** [reply]
Same problem happening on Ubuntu (I do not recall which one, probably one of the latest). Checkbox of remember settings is checked by default but ignored every time unless connecting recently. From settings, I also checked to share all screens but to no avail.

EDIT: installation method, debian package with dpkg -i downloaded from latest release in github.

**Reply by giarve** [reply]
Transfer file is working perfectly fine though.

but on connecting this happens:

![image](https://github.com/user-attachments/assets/4dc77e1c-6625-4784-b34f-a7a71b9145ad)


after closing the RustDesk window on client it goes offline (maybe crashed server-side?) and I cannot reconnect anymore. Will have to wait until I can physically check screen.

EDIT: systemd restart the service or something.

**Reply by giarve** [reply]
Could connect through ssh by tcp forwarding

systemd logs:

> Oct 30 14:12:40 server-VirtualBox rustdesk[3330598]: Cannot load libcuda.so.1
Oct 30 14:12:40 server-VirtualBox rustdesk[3330598]: Cannot load libcuda.so.1
Oct 30 14:12:56 server-VirtualBox rustdesk[3331862]: MESA: error: ZINK: failed to choose pdev
Oct 30 14:12:56 server-VirtualBox rustdesk[3331862]: libEGL warning: egl: failed to create dri2 screen
Oct 30 14:12:56 server-VirtualBox rustdesk[3331862]: flutter: launch args: [--cm]
Oct 30 14:12:56 server-VirtualBox rustdesk[3331862]: flutter: --cm started
Oct 30 14:12:56 server-VirtualBox rustdesk[3331862]: flutter: initializing FFI cm
Oct 30 14:12:56 server-VirtualBox rustdesk[3331862]: flutter: _appType:cm,info1-id:d79966d662e67961fe3d,info2-name:Ubuntu,dir:/home/server/Documents
Oct 30 14:12:56 server-VirtualBox rustdesk[3331862]: flutter: _globalFFI init
Oct 30 14:12:56 server-VirtualBox rustdesk[3331862]: flutter: registerEventHandler callback_query_onlines address book peer
Oct 30 14:12:56 server-VirtualBox rustdesk[3331862]: flutter: registerEventHandler load_address_book_peers address book peer
Oct 30 14:12:56 server-VirtualBox rustdesk[3331862]: flutter: registerEventHandler callback_query_onlines group peer
Oct 30 14:12:56 server-VirtualBox rustdesk[3331862]: flutter: registerEventHandler load_group_peers group peer
Oct 30 14:12:56 server-VirtualBox rustdesk[3331862]: flutter: registerEventHandler callback_query_onlines recent peer
Oct 30 14:12:56 server-VirtualBox rustdesk[3331862]: flutter: registerEventHandler load_recent_peers recent peer
Oct 30 14:12:56 server-VirtualBox rustdesk[3331862]: flutter: registerEventHandler callback_query_onlines fav peer
Oct 30 14:12:56 server-VirtualBox rustdesk[3331862]: flutter: registerEventHandler load_fav_peers fav peer
Oct 30 14:12:56 server-VirtualBox rustdesk[3331862]: flutter: registerEventHandler callback_query_onlines discovered peer
Oct 30 14:12:56 server-VirtualBox rustdesk[3331862]: flutter: registerEventHandler load_lan_peers discovered peer
Oct 30 14:12:56 server-VirtualBox rustdesk[3331862]: flutter: _globalFFI init end
Oct 30 14:12:56 server-VirtualBox rustdesk[3331862]: flutter: registerEventHandler theme theme
Oct 30 14:12:56 server-VirtualBox rustdesk[3331862]: flutter: registerEventHandler language language
Oct 30 14:12:56 server-VirtualBox rustdesk[3331862]: flutter: registerEventHandler native_ui native_ui
Oct 30 14:12:56 server-VirtualBox rustdesk[3331862]: flutter: Bring window 'null' on top
Oct 30 14:13:38 server-VirtualBox rustdesk[3331862]: flutter: model  closed
Oct 30 14:13:38 server-VirtualBox rustdesk[3331862]: gtk_widget_set_opacity: assertion 'GTK_IS_WIDGET (widget)' failed
Oct 30 14:13:38 server-VirtualBox rustdesk[3331862]: gtk_window_get_position: assertion 'GTK_IS_WINDOW (window)' failed
Oct 30 14:13:38 server-VirtualBox rustdesk[3331862]: gtk_window_get_size: assertion 'GTK_IS_WINDOW (window)' failed
Oct 30 14:13:38 server-VirtualBox rustdesk[3331862]: gtk_widget_hide: assertion 'GTK_IS_WIDGET (widget)' failed
Oct 30 14:13:38 server-VirtualBox rustdesk[3331862]: gtk_window_move: assertion 'GTK_IS_WINDOW (window)' failed
Oct 30 14:13:38 server-VirtualBox rustdesk[3331862]: gtk_window_resize: assertion 'GTK_IS_WINDOW (window)' failed
Oct 30 14:13:38 server-VirtualBox rustdesk[3331862]: void fl_view_set_textures(FlView *, GdkGLContext *, GPtrArray *): assertion 'FL_IS_VIEW(self)' failed
Oct 30 14:16:49 server-VirtualBox rustdesk[3350549]: MESA: error: ZINK: failed to choose pdev
Oct 30 14:16:49 server-VirtualBox rustdesk[3350549]: libEGL warning: egl: failed to create dri2 screen
Oct 30 14:16:49 server-VirtualBox rustdesk[3350549]: flutter: launch args: [--cm]
Oct 30 14:16:49 server-VirtualBox rustdesk[3350549]: flutter: --cm started
Oct 30 14:16:49 server-VirtualBox rustdesk[3350549]: flutter: initializing FFI cm
Oct 30 14:16:49 server-VirtualBox rustdesk[3350549]: flutter: _appType:cm,info1-id:d79966e67961fe3d,info2-name:Ubuntu,dir:/home/server/Documents
Oct 30 14:16:49 server-VirtualBox rustdesk[3350549]: flutter: _globalFFI init
Oct 30 14:16:49 server-VirtualBox rustdesk[3350549]: flutter: registerEventHandler callback_query_onlines address book peer
Oct 30 14:16:49 server-VirtualBox rustdesk[3350549]: flutter: registerEventHandler load_address_book_peers address book peer
Oct 30 14:16:49 server-VirtualBox rustdesk[3350549]: flutter: registerEventHandler callback_query_onlines group peer
Oct 30 14:16:49 server-VirtualBox rustdesk[3350549]: flutter: registerEventHandler load_group_peers group peer
Oct 30 14:16:49 server-VirtualBox rustdesk[3350549]: flutter: registerEventHandler callback_query_onlines recent peer
Oct 30 14:16:49 server-VirtualBox rustdesk[3350549]: flutter: registerEventHandler load_recent_peers recent peer
Oct 30 14:16:49 server-VirtualBox rustdesk[3350549]: flutter: registerEventHandler callback_query_onlines fav peer
Oct 30 14:16:49 server-VirtualBox rustdesk[3350549]: flutter: registerEventHandler load_fav_peers fav peer
Oct 30 14:16:49 server-VirtualBox rustdesk[3350549]: flutter: registerEventHandler callback_query_onlines discovered peer
Oct 30 14:16:49 server-VirtualBox rustdesk[3350549]: flutter: registerEventHandler load_lan_peers discovered peer
Oct 30 14:16:49 server-VirtualBox rustdesk[3350549]: flutter: _globalFFI init end
Oct 30 14:16:49 server-VirtualBox rustdesk[3350549]: flutter: registerEventHandler theme theme
Oct 30 14:16:49 server-VirtualBox rustdesk[3350549]: flutter: registerEventHandler language language
Oct 30 14:16:49 server-VirtualBox rustdesk[3350549]: flutter: registerEventHandler native_ui native_ui
Oct 30 14:16:50 server-VirtualBox rustdesk[3350549]: flutter: Bring window 'null' on top

**Comment by ic3man5** [comment]
This might be related: https://bugs.kde.org/show_bug.cgi?id=482594

**Comment by rustdesk** [comment]
https://rustdesk.com/blog/unattended-remote-access-wayland
